CorsixTH/Lua/announcer.lua

```diff
@@ -50,7 +44,7 @@ function AnnouncementQueue:push(priority, entry)
 end
 
 function AnnouncementQueue:pop()
-  for _, entries in pairs(self.priorities) do
+  for _, entries in ipairs(self.priorities) do
```

Comment:
Why did you end up changing this from pairs to ipairs?

Reply:
I've read (http://lua-users.org/wiki/ForTutorial) that the order of pairs is not defined, which is not what we want if critical announcements need to be played before low priority annoucements. ipairs iterates over the table in a defined numeric order.

Reply:
ipairs iterates in index order, but your table doesn't use numeric indexes so it's moot. If you need guaranteed order you will have to change it into an array instead.

```lua
  self.priorities = {
    AnnouncementPriority.Critical,
    AnnouncementPriority.High,
    AnnouncementPriority.Normal,
    AnnouncementPriority.Low
  }
```

Reply:
Which doesn't work at all ... keep another table (array) of the priorities in order and loop through that instead.

Reply:
... disregard everything, those are numbers. Sorry.